application/src/main/java/org/thingsboard/server/service/edge/rpc/processor/device/DeviceProfileEdgeProcessor.java

```diff
@@ -127,19 +129,24 @@ public DownlinkMsg convertDeviceProfileEventToDownlink(EdgeEvent edgeEvent, Edge
     }
 
     @Override
-    protected void setDefaultRuleChainId(TenantId tenantId, DeviceProfile deviceProfile, DeviceProfileUpdateMsg deviceProfileUpdateMsg) {
+    protected void setDefaultRuleChainId(TenantId tenantId, DeviceProfile deviceProfile) {
         // do nothing on cloud
     }
 
     @Override
-    protected void setDefaultEdgeRuleChainId(TenantId tenantId, DeviceProfile deviceProfile, DeviceProfileUpdateMsg deviceProfileUpdateMsg) {
-        UUID defaultEdgeRuleChainUUID = safeGetUUID(deviceProfileUpdateMsg.getDefaultRuleChainIdMSB(), deviceProfileUpdateMsg.getDefaultRuleChainIdLSB());
+    protected void setDefaultEdgeRuleChainId(DeviceProfile deviceProfile, RuleChainId ruleChainId, DeviceProfileUpdateMsg deviceProfileUpdateMsg, boolean isEdgeVersionDeprecated) {
+        UUID defaultEdgeRuleChainUUID = isEdgeVersionDeprecated
+                ? safeGetUUID(deviceProfileUpdateMsg.getDefaultRuleChainIdMSB(), deviceProfileUpdateMsg.getDefaultRuleChainIdLSB())
+                : deviceProfile.getDefaultRuleChainId() != null ? deviceProfile.getDefaultRuleChainId().getId() : null;
         deviceProfile.setDefaultEdgeRuleChainId(defaultEdgeRuleChainUUID != null ? new RuleChainId(defaultEdgeRuleChainUUID) : null);
+        deviceProfile.setDefaultRuleChainId(ruleChainId);
```

Comment:
probably it should be moved to:
**protected void setDefaultRuleChainId(TenantId tenantId, DeviceProfile deviceProfile) {**
?